### DISCUSSION TITLE
Where to find "lib/gen" ?

### DISCUSSION THREAD

**Discussion by Donald2010** [windmill-labs/windmill#4230] [Questions]
In some file, like "frontend/src/lib/components/details/utils.ts", there is import command: `import { ScriptService, JobService, type Script } from '$lib/gen'`. but I can't find the gen.ts file. Is it auto genereated from somewhere ?

**Comment by pyranota** [comment]
Yes it is autogenerated by `npm run generate-backend-client`. Process will read openapi specification and generate ts wrapper around it which can be used by frontend.
